distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1474,6 +1538,65 @@ public void testProductDownloadListWithOneActiveDownload() {
         assertThat((List<String>) download.get("users"), contains("Guest@Guest@127.0.0.1"));
     }
 
+    @Test
+    public void testCancelDownload() throws Exception {
+        getSecurityPolicy().configureWebContextPolicy(null, "/=SAML|basic,/solr=SAML|PKI|basic",
+                null, null);
+        localhostCometDClient = setupCometDClientWithUser(
+                Arrays.asList(NOTIFICATIONS_CHANNEL, ACTIVITIES_CHANNEL), "localhost", "localhost");
+        String filename = testName + ".txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+
+        cswServer.whenHttp()
+                .match(post("/services/csw"), withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId))
+                .then(ok(), contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(200), 0));
+
+        String startDownloadUrl = RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl() + "?source="
+                + CSW_STUB_SOURCE_ID + "&metacard=" + metacardId;
+
```

Comment:
In my PR (https://github.com/codice/ddf/pull/1009/files) I created a DownloadHandle class that could probably be used here as well to set up lines 1551 through 1565.